src/api/views/user_settings.py

```
@@ -75,31 +65,28 @@ def validate_value(self, value, value_type):
             'list': list,
         }
 
-        # if list, check if list of strings
-        if isinstance(value, list):
-            if not value or not all(
-                isinstance(i, str) for i in value
-            ):  # must be a list of strings
-                return False
-
-        if value_type not in value_type_mapping.keys():
+        # in case of lists only lists of strings are allowed
+        if isinstance(value, list) and (
+            not value or not all(isinstance(i, str) for i in value)
+        ):
             return False
 
-        if not isinstance(value, value_type_mapping[value_type]):
+        if value_type not in value_type_mapping:
```

I think we should do this check first in the function, and maybe raise an exception instead of just returning `False`.

---

Well, the whole `validate_value` method in the ViewSet seems kind of misplaced. Why not use the serializer for that? I guess this is maybe an issue of legacy code, which we never really reviewed in cas?
But I would like to be able to close this PR before the long holiday, so I created a separate ticket for it, in which we could also discuss, whether and how to go about it: https://github.com/base-angewandte/baseauth/issues/40